docs/api/README.md

```diff
@@ -349,9 +349,7 @@ You can also pass in a channel as argument and the behavior is the same as [take
 
 ### `put(action)`
 
-Creates an Effect description that instructs the middleware to dispatch an action to the Store.
-This effect is non-blocking and any errors that are thrown downstream (e.g. in a reducer) will
-not bubble back into the saga.
+Creates an Effect description that instructs the middleware to dispatch an action to the Store. This is similar to calling `dispatch` directly on the action; your Store will be synchronously updated unless you have other middlewares with asynchronous flows that delay the propagation of the action. Downstream errors will be bubbled up.
```

Comment:
after consideration - `put` is not entirely synchronous, it gets scheduled in internal queue and is guaranteed to be dispatched within current stack frame, but this might not happen immediately. This is an important part of the behavior and has to be mentioned here.

Reply:
To be candid I'm not quite sure what "immediate" means here, would you mind clarifying?

If the dispatch is done in the current stack frame and the store is updated before the next line of code in the saga function is executed, isn't that considered synchronous? 🤔

Reply:
Well, "nested" puts are kinda flattened/queued. Consider this:
```js
yield put(FOO)

// other saga
yield takeEvery(FOO, function*() {
  // this gets called synchronously in response to `put(FOO)`
  // this BAR dispatch won't happen immediately though
  // it will get called after we exit FOO processing
  yield put(BAR)
})
```